Checking if there is a collider when calling `SoftBody::remove_collision_exception_with`

Duddino opened the pull request:
fixes #46651

pouleyKetchoupp reviewed (changes requested):
Thanks for your contribution! Although instead of raising an error, this case could be handled properly in the call to `remove_collision_exception` the same way it already does for `add_collision_exception`.

This is the code for `add_collision_exception`:
https://github.com/godotengine/godot/blob/17e66382b8e50cf7b58b8ecf4c12628408529538/modules/bullet/collision_object_bullet.cpp#L138-L147

The same check on `bt_collision_object` could be done here:
https://github.com/godotengine/godot/blob/17e66382b8e50cf7b58b8ecf4c12628408529538/modules/bullet/collision_object_bullet.cpp#L149-L155

And this method could also have a similar check:
https://github.com/godotengine/godot/blob/17e66382b8e50cf7b58b8ecf4c12628408529538/modules/bullet/collision_object_bullet.cpp#L161-L168

And finally this one could benefit from a change to make it safe:
https://github.com/godotengine/godot/blob/17e66382b8e50cf7b58b8ecf4c12628408529538/modules/bullet/collision_object_bullet.cpp#L157-L160

It could be replaced with:
```
bool CollisionObjectBullet::has_collision_exception(const CollisionObjectBullet *p_otherCollisionObject) const { 
 	return exceptions.has(p_otherCollisionObject->get_self());
 } 
```
It should be the exact same result but this way it wouldn't rely on `bt_collision_object` being valid.

One last piece of advice is, since Bullet is disabled on master at the moment (it's not compiled at all), make sure to test on the 3.2 branch locally (this PR can still be on master though).

Commit:
Added some checks to prevent accessing a null collider
Previously godot would try to access
`CollisionObjectBullet::bt_collision_object` even if it was null.
Fixes #46651

akien-mga commented:
Thanks!

akien-mga commented:
Cherry-picked for 3.2.4.
